File: lib_js/models/supabase/panelViews.js
import { SupabaseModel } from "./supabaseModel.js";

// Define ViewUserPanelDiscussionModel
export class ViewUserPanelDiscussionModel extends SupabaseModel {
    static TABLE_NAME = "view_user_panel_discussion";
    static TABLE_FIELDS = {
        userDataId: { type: "uuid", required: false, dbColumn: "user_data_id" },
        panelDiscussionId: {
            type: "uuid",
            required: false,
            dbColumn: "panel_discussion_id"
        },
        authId: { type: "uuid", required: false, dbColumn: "auth_id" },
        title: { type: "string", required: false, dbColumn: "title" },
        tags: { type: "array", required: false, dbColumn: "tags" },
        item: { type: "string", required: false, dbColumn: "item" },
        userDataData: {
            type: "json",
            required: false,
            dbColumn: "user_data_data"
        },
        panelDiscussionMetadata: {
            type: "json",
            required: false,
            dbColumn: "panel_discussion_metadata"
        },
        userDataCreatedAt: {
            type: "date",
            required: false,
            dbColumn: "user_data_created_at"
        },
        userDataUpdatedAt: {
            type: "date",
            required: false,
            dbColumn: "user_data_updated_at"
        }
    };
}

// Define ViewUserPanelTranscriptModel
export class ViewUserPanelTranscriptModel extends SupabaseModel {
    static TABLE_NAME = "view_user_panel_transcript";
    static TABLE_FIELDS = {
        userDataId: { type: "uuid", required: false, dbColumn: "user_data_id" },
        panelTranscriptId: {
            type: "uuid",
            required: false,
            dbColumn: "panel_transcript_id"
        },
        authId: { type: "uuid", required: false, dbColumn: "auth_id" },
        item: { type: "string", required: false, dbColumn: "item" },
        title: { type: "string", required: false, dbColumn: "title" },
        lang: { type: "string", required: false, dbColumn: "lang" },
        transcript: { type: "json", required: false, dbColumn: "transcript" },
        file: { type: "string", required: false, dbColumn: "file" },
        bucketId: { type: "string", required: false, dbColumn: "bucket_id" },
        type: { type: "string", required: false, dbColumn: "type" },
        userDataData: {
            type: "json",
            required: false,
            dbColumn: "user_data_data"
        },
        panelTranscriptMetadata: {
            type: "json",
            required: false,
            dbColumn: "panel_transcript_metadata"
        },
        userDataCreatedAt: {
            type: "date",
            required: false,
            dbColumn: "user_data_created_at"
        },
        userDataUpdatedAt: {
            type: "date",
            required: false,
            dbColumn: "user_data_updated_at"
        }
    };
}

// Define ViewUserPanelAudioModel
export class ViewUserPanelAudioModel extends SupabaseModel {
    static TABLE_NAME = "view_user_panel_audio";
    static TABLE_FIELDS = {
        userDataId: { type: "uuid", required: false, dbColumn: "user_data_id" },
        panelAudioId: {
            type: "uuid",
            required: false,
            dbColumn: "panel_audio_id"
        },
        authId: { type: "uuid", required: false, dbColumn: "auth_id" },
        item: { type: "string", required: false, dbColumn: "item" },
        title: { type: "string", required: false, dbColumn: "title" },
        lang: { type: "string", required: false, dbColumn: "lang" },
        file: { type: "string", required: false, dbColumn: "file" },
        bucketId: { type: "string", required: false, dbColumn: "bucket_id" },
        userDataData: {
            type: "json",
            required: false,
            dbColumn: "user_data_data"
        },
        panelAudioMetadata: {
            type: "json",
            required: false,
            dbColumn: "panel_audio_metadata"
        },
        userDataCreatedAt: {
            type: "date",
            required: false,
            dbColumn: "user_data_created_at"
        },
        userDataUpdatedAt: {
            type: "date",
            required: false,
            dbColumn: "user_data_updated_at"
        }
    };
}

File: source/models/supabase/panel_views.py
from typing import ClassVar, Dict, Optional
from uuid import UUID
from pydantic import Field
from datetime import datetime
from source.models.supabase.supabase_model import SupabaseModel


class ViewUserPanelDiscussionModel(SupabaseModel):
    TABLE_NAME: ClassVar[str] = "view_user_panel_discussion"
    user_data_id: Optional[UUID] = Field(default=None)
    panel_discussion_id: Optional[UUID] = Field(default=None)
    auth_id: Optional[UUID] = Field(default=None)
    title: Optional[str] = Field(default=None)
    tags: Optional[list[str]] = Field(default=None)
    item: Optional[str] = Field(default=None)
    user_data_data: Optional[Dict] = Field(default=None)
    panel_discussion_metadata: Optional[Dict] = Field(default=None)
    user_data_created_at: Optional[datetime] = Field(default=None)
    user_data_updated_at: Optional[datetime] = Field(default=None)


class ViewUserPanelTranscriptModel(SupabaseModel):
    TABLE_NAME: ClassVar[str] = "view_user_panel_transcript"
    user_data_id: Optional[UUID] = Field(default=None)
    panel_transcript_id: Optional[UUID] = Field(default=None)
    auth_id: Optional[UUID] = Field(default=None)
    item: Optional[str] = Field(default=None)
    title: Optional[str] = Field(default=None)
    lang: Optional[str] = Field(default=None)
    transcript: Optional[Dict] = Field(default=None)
    file: Optional[str] = Field(default=None)
    bucket_id: Optional[str] = Field(default=None)
    type: Optional[str] = Field(default=None)
    user_data_data: Optional[Dict] = Field(default=None)
    panel_transcript_metadata: Optional[Dict] = Field(default=None)
    user_data_created_at: Optional[datetime] = Field(default=None)
    user_data_updated_at: Optional[datetime] = Field(default=None)


class ViewUserPanelAudioModel(SupabaseModel):
    TABLE_NAME: ClassVar[str] = "view_user_panel_audio"
    user_data_id: Optional[UUID] = Field(default=None)
    panel_audio_id: Optional[UUID] = Field(default=None)
    auth_id: Optional[UUID] = Field(default=None)
    item: Optional[str] = Field(default=None)
    title: Optional[str] = Field(default=None)
    lang: Optional[str] = Field(default=None)
    file: Optional[str] = Field(default=None)
    bucket_id: Optional[str] = Field(default=None)
    user_data_data: Optional[Dict] = Field(default=None)
    panel_audio_metadata: Optional[Dict] = Field(default=None)
    user_data_created_at: Optional[datetime] = Field(default=None)
    user_data_updated_at: Optional[datetime] = Field(default=None)
